presto-main/src/main/java/com/facebook/presto/sql/analyzer/ExpressionAnalyzer.java

```diff
@@ -622,9 +622,10 @@ protected Type visitLikePredicate(LikePredicate node, StackableAstVisitorContext
 
             Type patternType = getVarcharType(node.getPattern(), context);
             coerceType(context, node.getPattern(), patternType, "Pattern for LIKE expression");
-            if (node.getEscape() != null) {
-                Type escapeType = getVarcharType(node.getEscape(), context);
-                coerceType(context, node.getEscape(), escapeType, "Escape for LIKE expression");
+            if (node.getEscape().isPresent()) {
+                Expression escape = node.getEscape().get();
+                Type escapeType = getVarcharType(escape, context);
+                coerceType(context, escape, escapeType, "Escape for LIKE expression");
```

Comment:
```
node.getEscape().ifPresent(escape -> coerceType(context, escape, getVarcharType(escape, context), "Escape for LIKE expression"))
```
?

Reply:
I think that makes the code less readable. That single line ends up doing too much and makes it harder to spot that there's a side-effect implied by ifPresent